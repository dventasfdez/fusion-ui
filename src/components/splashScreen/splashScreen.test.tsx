import React from "react";
import SplashScreen from "./splashScreen";
import "@testing-library/jest-dom/extend-expect";
import renderer, { act } from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react";

const backgroundImageURI = "https://www.w3schools.com/html/img_girl.jpg";
const backgroundVideoURI = "https://assets.mixkit.co/videos/preview/mixkit-clouds-in-the-sky-flowing-with-the-wind-21584-large.mp4";

const SplashScreenTest = (props?: any) => {
  return <SplashScreen {...props} logo="https://upload.wikimedia.org/wikipedia/commons/thumb/9/92/Adecco_logo.svg/800px-Adecco_logo.svg.png" />;
};
const SplashScreenTestLogoObject = (props?: any) => {
  return <SplashScreen {...props} logo={<img src="https://upload.wikimedia.org/wikipedia/commons/thumb/9/92/Adecco_logo.svg/800px-Adecco_logo.svg.png" className="img" />} className="stepone-ui" />;
};

it("splash screen component should render", () => {
  const component = renderer.create(<SplashScreenTest data-testid="splash-screen" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("splash screen component without test id should render", () => {
  const component = renderer.create(<SplashScreenTest />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("splash screen accent component should render", () => {
  const component = renderer.create(<SplashScreenTest data-testid="splash-screen" accent />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("splash screen accent component on mobile should render", () => {
  window = Object.assign(window, { innerWidth: 600 });
  const component = renderer.create(<SplashScreenTest data-testid="splash-screen" accent />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("splash screen image component should render", () => {
  const component = renderer.create(<SplashScreenTest data-testid="splash-screen" image={backgroundImageURI} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("splash screen video component should render", () => {
  const component = renderer.create(<SplashScreenTest data-testid="splash-screen" video={backgroundVideoURI} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("splash screen with logo object", () => {
  const component = renderer.create(<SplashScreenTestLogoObject />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
it("splash screen with logo object", () => {
  const component = renderer.create(<SplashScreenTestLogoObject data-testid="splash-screen" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
it("splash screen with logo object and image", () => {
  const component = renderer.create(<SplashScreenTestLogoObject image={backgroundImageURI} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
it("splash screen with logo object and video", () => {
  const component = renderer.create(<SplashScreenTestLogoObject video={backgroundVideoURI} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe("test resize in splash screen", () => {
  it("small screen", async () => {
    window.innerWidth = 600;
    window.innerHeight = 600;
    fireEvent(window, new Event("resize"));

    const { queryAllByTestId } = render(<SplashScreenTest data-testid="splash-screen" />);
    expect(queryAllByTestId("splash-screen").length).toBe(1);
  });
  it("big screen", async () => {
    window.innerWidth = 1000;
    window.innerHeight = 1000;
    const { queryAllByTestId } = render(<SplashScreenTest data-testid="splash-screen" />);
    fireEvent(window, new Event("resize"));

    expect(queryAllByTestId("splash-screen").length).toBe(0);
  });
  it("small screen", async () => {
    window.innerWidth = 600;
    const { queryAllByTestId } = render(<SplashScreenTest data-testid="splash-screen" />);
    expect(queryAllByTestId("splash-screen").length).toBe(1);
  });
  it("splash screen with icon component", () => {
    const component = renderer.create(<SplashScreen data-testid="splash-screen" logo={<div>icon</div>} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("splash screen with component icon and no data-testid", () => {
    const component = renderer.create(<SplashScreen logo="string" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
