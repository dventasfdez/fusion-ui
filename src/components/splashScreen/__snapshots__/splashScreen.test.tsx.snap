// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`splash screen accent component on mobile should render 1`] = `
<div
  className="splash-screen_accent "
  data-testid="splash-screen"
>
  <img
    alt="logo"
    className="splash-screen-logo"
    data-testid="splash-screen-logo-image"
    src="https://upload.wikimedia.org/wikipedia/commons/thumb/9/92/Adecco_logo.svg/800px-Adecco_logo.svg.png"
  />
</div>
`;

exports[`splash screen accent component should render 1`] = `null`;

exports[`splash screen component should render 1`] = `null`;

exports[`splash screen component without test id should render 1`] = `null`;

exports[`splash screen image component should render 1`] = `null`;

exports[`splash screen video component should render 1`] = `
<div
  className="splash-screen "
  data-testid="splash-screen"
>
  <div
    className="splash-screen-video"
    data-testid="splash-screen-video-wrapper"
  >
    <video
      autoPlay={true}
      loop={true}
      muted={true}
    >
      <source
        src="https://assets.mixkit.co/videos/preview/mixkit-clouds-in-the-sky-flowing-with-the-wind-21584-large.mp4"
        type="video/mp4"
      />
    </video>
  </div>
  <img
    alt="logo"
    className="splash-screen-logo"
    data-testid="splash-screen-logo-image"
    src="https://upload.wikimedia.org/wikipedia/commons/thumb/9/92/Adecco_logo.svg/800px-Adecco_logo.svg.png"
  />
</div>
`;

exports[`splash screen with logo object 1`] = `null`;

exports[`splash screen with logo object 2`] = `
<div
  className="splash-screen.stepone-ui"
  data-testid="splash-screen"
>
  <img
    className="img splash-screen-logo"
    data-testid="splash-screen-logo-wrapper"
    src="https://upload.wikimedia.org/wikipedia/commons/thumb/9/92/Adecco_logo.svg/800px-Adecco_logo.svg.png"
  />
</div>
`;

exports[`splash screen with logo object and image 1`] = `null`;

exports[`splash screen with logo object and video 1`] = `
<div
  className="splash-screen.stepone-ui"
>
  <div
    className="splash-screen-video"
  >
    <video
      autoPlay={true}
      loop={true}
      muted={true}
    >
      <source
        src="https://assets.mixkit.co/videos/preview/mixkit-clouds-in-the-sky-flowing-with-the-wind-21584-large.mp4"
        type="video/mp4"
      />
    </video>
  </div>
  <img
    className="img splash-screen-logo"
    src="https://upload.wikimedia.org/wikipedia/commons/thumb/9/92/Adecco_logo.svg/800px-Adecco_logo.svg.png"
  />
</div>
`;

exports[`test resize in splash screen splash screen with component icon and no data-testid 1`] = `
<div
  className="splash-screen "
>
  <img
    alt="logo"
    className="splash-screen-logo"
    src="string"
  />
</div>
`;

exports[`test resize in splash screen splash screen with icon component 1`] = `null`;
